From 5e5ef8c17624f742bbfb5f4070abbaeebd3608d6 Mon Sep 17 00:00:00 2001
From: Mirko Biselli <mirkox.biselli@intel.com>
Date: Fri, 18 Jan 2013 15:52:49 +0100
Subject: [PORT FROM MAIN][BT] factorization for libbt-vendor

BZ: 81384

libbt-vendor module is now dependent on the BT chip used on each
platform, so the source files change depending on COMBO_CHIP_VENDOR
definition.

Change-Id: I08f7a16be0a8798f2291a92e730820c1e85c3c9f
Orig-Change-Id: I87275259ea486f288998c67ad919c8f02221165b
Feature: BT
Category: Integration
Purpose: ABSP-Adapter
Origin: Internal
Upstream-candidate: no
Signed-off-by: Mirko Biselli <mirkox.biselli@intel.com>
Reviewed-on: http://android.intel.com:8080/87540
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libbt/Android.mk |   51 ++++++++++++++++++++++++++++++++++++---------------
 1 files changed, 36 insertions(+), 15 deletions(-)

diff --git a/libbt/Android.mk b/libbt/Android.mk
index 3ce6fc7..03720fa 100644
--- a/libbt/Android.mk
+++ b/libbt/Android.mk
@@ -4,30 +4,51 @@ include $(CLEAR_VARS)
 
 BDROID_DIR := $(TOP_DIR)external/bluetooth/bluedroid
 
-LOCAL_SRC_FILES := \
-        src/bt_vendor_brcm.c \
-        src/hardware.c \
-        src/userial_vendor.c \
-        src/upio.c \
-        src/conf.c
-
-LOCAL_C_INCLUDES += \
-        $(LOCAL_PATH)/include \
-        $(BDROID_DIR)/hci/include
-
-LOCAL_SHARED_LIBRARIES := \
-        libcutils
+include $(TOP_DIR)vendor/intel/common/ComboChipVendor.mk
 
 LOCAL_MODULE := libbt-vendor
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_CLASS := SHARED_LIBRARIES
-LOCAL_MODULE_OWNER := broadcom
 LOCAL_MODULE_PATH := $(TARGET_OUT_VENDOR_SHARED_LIBRARIES)
 
-include $(LOCAL_PATH)/vnd_buildcfg.mk
+# BCM configuration
+ifeq ($(COMBO_CHIP_VENDOR), bcm)
+    LOCAL_C_INCLUDES := \
+        $(BDROID_DIR)/hci/include \
+        $(LOCAL_PATH)/include
+    LOCAL_SRC_FILES := \
+        src/bt_vendor_brcm.c \
+        src/hardware.c \
+        src/userial_vendor.c \
+        src/upio.c \
+        src/conf.c
+    LOCAL_SHARED_LIBRARIES := libcutils
+    LOCAL_MODULE_OWNER := broadcom
+    include $(LOCAL_PATH)/vnd_buildcfg.mk
+endif
+# end of BCM configuration
+
+
+# TI configuration
+ifeq ($(COMBO_CHIP_VENDOR), ti)
+    LOCAL_PATH := $(ANDROID_BUILD_TOP)
+    TI_BT_VENDOR_PATH := hardware/ti/wpan/bluedroid_wilink
+    LOCAL_C_INCLUDES := $(BDROID_DIR)/hci/include
+    LOCAL_SRC_FILES := $(TI_BT_VENDOR_PATH)/libbt-vendor-ti.c
+    LOCAL_SHARED_LIBRARIES := \
+        libcutils \
+        libnativehelper \
+        libutils \
+        liblog
+    LOCAL_MODULE_OWNER := ti
+endif
+# end of TI configuration
 
 include $(BUILD_SHARED_LIBRARY)
 
+# LOCAL_PATH needs to be redefine in case TI configuration is used
+LOCAL_PATH := $(call my-dir)
+
 ifeq ($(TARGET_PRODUCT), full_maguro)
     include $(LOCAL_PATH)/conf/samsung/maguro/Android.mk
 endif
-- 
1.7.4.1

